crates/rust-client/src/store/data_store.rs

```diff
@@ -23,110 +19,94 @@ use crate::store::StoreError;
 pub(crate) struct ClientDataStore {
     /// Local database containing information about the accounts managed by this client.
     pub(crate) store: alloc::sync::Arc<dyn Store>,
+    /// Store used to provide MAST nodes to the transaction executor.
+    pub transaction_mast_store: Arc<TransactionMastStore>,
```

Comment:
nit: Since `ClientDataStore` is `pub(crate)` this `pub` works as a `pub(crate)` to. Maybe having it explicitly is better for readability.
```suggestion
    pub(crate) transaction_mast_store: Arc<TransactionMastStore>,
```